File: frontend/src/api/userProviderKeyController.ts
// @ts-ignore
/* eslint-disable */
import request from '@/request'

/** 添加用户提供者密钥 POST /api/byok/add */
export async function addUserProviderKey(
  body: API.UserProviderKeyAddRequest,
  options?: { [key: string]: any }
) {
  return request<API.ResultBoolean>('/api/byok/add', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 删除用户提供者密钥 POST /api/byok/delete */
export async function deleteUserProviderKey(
  body: API.DeleteRequest,
  options?: { [key: string]: any }
) {
  return request<API.ResultBoolean>('/api/byok/delete', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

/** 获取我的提供者密钥列表 GET /api/byok/my/list */
export async function listMyProviderKeys(options?: { [key: string]: any }) {
  return request<API.ResultListUserProviderKeyVO>('/api/byok/my/list', {
    method: 'GET',
    ...(options || {}),
  })
}

/** 更新用户提供者密钥 POST /api/byok/update */
export async function updateUserProviderKey(
  body: API.UserProviderKeyUpdateRequest,
  options?: { [key: string]: any }
) {
  return request<API.ResultBoolean>('/api/byok/update', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    data: body,
    ...(options || {}),
  })
}

File: backend/src/main/java/com/lilac/controller/UserProviderKeyController.java
package com.lilac.controller;

import com.lilac.annotation.AuthCheck;
import com.lilac.common.DeleteRequest;
import com.lilac.constant.UserConstant;
import com.lilac.domain.dto.byok.UserProviderKeyAddRequest;
import com.lilac.domain.dto.byok.UserProviderKeyUpdateRequest;
import com.lilac.domain.entity.User;
import com.lilac.domain.result.Result;
import com.lilac.domain.vo.UserProviderKeyVO;
import com.lilac.enums.HttpsCodeEnum;
import com.lilac.exception.BusinessException;
import com.lilac.service.UserProviderKeyService;
import com.lilac.service.UserService;
import jakarta.annotation.Resource;
import jakarta.servlet.http.HttpServletRequest;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * 用户提供者密钥控制类
 */
@Slf4j
@RestController
@RequestMapping("/byok")
public class UserProviderKeyController {

    @Resource
    private UserProviderKeyService userProviderKeyService;
    @Resource
    private UserService userService;

    /**
     * 添加用户提供者密钥
     */
    @PostMapping("/add")
    @AuthCheck(mustRole = UserConstant.DEFAULT_ROLE)
    public Result<Boolean> addUserProviderKey(@RequestBody UserProviderKeyAddRequest request, HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        boolean result = userProviderKeyService.addUserProviderKey(request, loginUser.getId());
        return Result.success(result);
    }

    /**
     * 更新用户提供者密钥
     */
    @PostMapping("/update")
    @AuthCheck(mustRole = UserConstant.DEFAULT_ROLE)
    public Result<Boolean> updateUserProviderKey(@RequestBody UserProviderKeyUpdateRequest request, HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        boolean result = userProviderKeyService.updateUserProviderKey(request, loginUser.getId());
        return Result.success(result);
    }

    /**
     * 删除用户提供者密钥
     */
    @PostMapping("/delete")
    @AuthCheck(mustRole = UserConstant.DEFAULT_ROLE)
    public Result<Boolean> deleteUserProviderKey(@RequestBody DeleteRequest request, HttpServletRequest httpRequest) {
        if (request == null || request.getId() == null) {
            throw new BusinessException(HttpsCodeEnum.PARAMS_ERROR);
        }
        User loginUser = userService.getLoginUser(httpRequest);
        boolean result = userProviderKeyService.deleteUserProviderKey(request.getId(), loginUser.getId());
        return Result.success(result);
    }

    /**
     * 获取我的提供者密钥列表
     */
    @GetMapping("/my/list")
    @AuthCheck(mustRole = UserConstant.DEFAULT_ROLE)
    public Result<List<UserProviderKeyVO>> listMyProviderKeys(HttpServletRequest httpRequest) {
        User loginUser = userService.getLoginUser(httpRequest);
        List<UserProviderKeyVO> list = userProviderKeyService.listUserProviderKeys(loginUser.getId());
        return Result.success(list);
    }
}